src/eventsAndStorageSelector.ts

```
@@ -0,0 +1,90 @@
+import { EventHandlerContext } from './processor'
+import * as foucocoEvents from './types/foucoco/events'
+import * as pendulumEvents from './types/pendulum/events'
+import * as amplitudeEvents from './types/amplitude/events'
+
+import * as foucocoStorage from './types/foucoco/storage'
+import * as pendulumStorage from './types/pendulum/storage'
+import * as amplitudeStorage from './types/amplitude/storage'
+
+import { StorageType as StorageTypeFoucoco } from './types/foucoco/support'
+import { StorageType as StorageTypePendulum } from './types/pendulum/support'
+import { StorageType as StorageTypeAmplitude } from './types/amplitude/support'
+// Must keep up to date with the possible versions
+// of events and storage
+const foucocoEventVersions: string[] = ['v1', 'v4']
+const foucocoStorageVersions: string[] = ['v1', 'v4']
+const pendulumEventVersions: string[] = ['v1', 'v3', 'v9', 'v10']
+const pendulumStorageVersions: string[] = ['v1', 'v3', 'v9', 'v10']
+const amplitudeEventVersions: string[] = ['v1', 'v3', 'v7', 'v8', 'v10', 'v12']
+const amplitudeStorageVersions: string[] = [
+    'v1',
+    'v3',
+    'v7',
+    'v8',
+    'v10',
+    'v12',
+]
+
+export function decodeEvent(
+    network: string,
+    ctx: EventHandlerContext,
+    moduleName: string,
+    eventName: string
+) {
+    let networkEventsAny
+    let eventVersions: string[]
+
+    if (network === 'foucoco') {
+        networkEventsAny = foucocoEvents as { [key: string]: any }
+        eventVersions = foucocoEventVersions
+    } else if (network === 'pendulum') {
+        networkEventsAny = pendulumEvents as { [key: string]: any }
+        eventVersions = pendulumEventVersions
+    } else {
+        networkEventsAny = amplitudeEvents as { [key: string]: any }
+        eventVersions = amplitudeEventVersions
+    }
```

Instead of returning amplitude by default, we could add another `else if` branch for amplitude and throw an error in the return. This is super minor and should never happen but if for example one day we introduce a typo in one of the `network` env variables, then this check would catch that error instead of silently decoding the events for amplitude.